DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceRegistryRepositoryTests.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Exceptions;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Helpers;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Models;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.Exceptions;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.Repository;
+using Moq;
+using Ploeh.AutoFixture;
+using Xunit;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.UnitTests.Infrastructure
+{
+    public class DeviceRegistryRepositoryTests
+    {
+        private readonly Fixture _fixture;
+        private readonly DeviceRegistryRepository _deviceRegistryRepository;
+        private readonly Mock<IDocumentDBClient<DeviceModel>> _mockDocumentDBClient;
+
+        public DeviceRegistryRepositoryTests()
+        {
+            this._fixture = new Fixture();
+            this._mockDocumentDBClient = new Mock<IDocumentDBClient<DeviceModel>>();
+            this._deviceRegistryRepository = new DeviceRegistryRepository(this._mockDocumentDBClient.Object);
+        }
+
+       
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public async Task GetDeviceAsync_should_throw_argument_exception_when_device_id_is_invalid(string deviceId)
+        {
+            await Assert.ThrowsAsync<ArgumentException>(() => this._deviceRegistryRepository.GetDeviceAsync(deviceId));
+        }
+
+        [Fact]
+        public async Task GetDeviceAsync_should_get_device_from_client()
+        {
+            var devices = this._fixture.Create<List<DeviceModel>>();
+            var deviceId = devices.First().DeviceProperties.DeviceID;
+
+            this._mockDocumentDBClient
+                .Setup(x => x.QueryAsync())
+                .ReturnsAsync(devices.AsQueryable());
+
+            var device = await this._deviceRegistryRepository.GetDeviceAsync(deviceId);
+
+            Assert.NotNull(device);
+            Assert.Same(devices[0], device);
```

Comment:
we should use Assert.Same in some previous tests as well as it does reference matching and not content matching